src/hypothesis/internal/summation.py

```diff
@@ -0,0 +1,45 @@
+# coding=utf-8
+#
+# This file is part of Hypothesis, which may be found at
+# https://github.com/HypothesisWorks/hypothesis-python
+#
+# Most of this work is copyright (C) 2013-2017 David R. MacIver
+# (david@drmaciver.com), but it contains contributions by others. See
+# CONTRIBUTING.rst for a full list of people who may hold copyright, and
+# consult the git log if you need to determine who owns an individual
+# contribution.
+#
+# This Source Code Form is subject to the terms of the Mozilla Public License,
+# v. 2.0. If a copy of the MPL was not distributed with this file, You can
+# obtain one at http://mozilla.org/MPL/2.0/.
+#
+# END HEADER
+
+from __future__ import division, print_function, absolute_import
+
+
+class KahanSummation(object):
+    """Incremental implementation of Kahan Summation.
+
+    Maintains a more accurate running sum of a series of values than we would
```

Comment:
nit: is it worth clarifying “a series of *floating point* values”?

Integer addition is arbitrary precision, at least in Python, I think?

Reply:
Yep - the algorithm only makes sense for values with varying absolute precision, ie some form of floating point, because for exact integers the carry value will always be zero.  The only exception is over/underflow, which can only happen for the Python2 `int` type.

That said, I'd just keep a list of values and use `math.fsum`!